File: added search/cdr/src/main/java/com/capstone/cdr/dto/CSVCallObject.java
package com.capstone.cdr.dto;

import com.capstone.cdr.entity.Type;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter

public class CSVCallObject {
    int id;
    String subsciberName;
    String subscriberNumber;
    String recieverName;
    String recieverNumber;
    String date;
    String time;
    String subscriberLocation;
    String recieverLocation;
    Type type;
    int duration;
    boolean isFailed;
    boolean hasVoiceMail;
    int voiceMailDuration; 
    String reason;   
}

File: added search/src/app/CSVCallObject.ts

import { AppType } from "src/AppType";

export interface CSVCallObject
{
    id: number;
    subscriberName: string;
    subscriberNumber: string;
    recieverName: string;
    recieverNumber: string;
    date: string;
    time: string;
    subscriberLocation: string;
    recieverLocation: string;
    type: AppType;
    duration: number;
    isFailed: boolean;
    hasVoiceMail: boolean;
    voiceMailDuration: number
}

                